### ISSUE TITLE
CPU Intensity Spike Issue with VST3 Plugins in Plugdata

### ISSUE THREAD

**Issue by WTEVR25674** [plugdata-team/plugdata#1972]
Hey Timothy Schoen,

I hope you're doing well. I recently started using Plugdata to run Ewan Bristow's plugin EB-FreakyTable for my music production experiments. However, I've noticed that VST3 plugins become quite CPU-intensive when used within Plugdata in my DAW. This seems to be related to a lack of CPU optimization in Plugdata, causing significant CPU spikes.

This issue may also affect other users. Could you please look into optimizing Plugdata for smoother performance? It would greatly benefit myself and other music producers. Thank you very much for your attention to this matter.

Best regards,
A concerned user.

**Comment by dromer** [comment]
"VST3 Plugins in Plugdata" << do you mean using the `[vstplugin~]` external?

Afaik this is not supported.

**Comment by WTEVR25674** [comment]
> "VST3 Plugins in Plugdata" << do you mean using the `[vstplugin~]` external?
> 
> Afaik this is not supported.

Yes, I’m referring to using external VST plugins inside Plugdata when it’s used within my DAW. I tried running the VST plugin EB-FreakyTable, which is an external VST3 plugin, and while it works inside Plugdata, it causes my CPU to spike quite heavily inside my DAW. This makes the plugin unenjoyable to use, not only for myself but potentially for other users and music producers as well.

**Comment by dromer** [comment]
`EB-FreakyTable` is not an external VST3 plugin, it is a PD patch.

**Comment by WTEVR25674** [comment]
> `EB-FreakyTable` is not an external VST3 plugin, it is a PD patch.

I apologize for the misunderstanding, you’re right, it's indeed a PD patch. However, the plugin Plugdata itself seems to cause a high CPU overload spike when running PD patches like EB-FreakyTable inside of it.

**Comment by dromer** [comment]
While there are certain things in plugdata that can be optimized, more likely than not it's your computer that cannot keep up.
